packages/app/models/stack-order/stack-order.ts

```diff
@@ -1,14 +1,20 @@
 import { StackOrder } from "@/models/stack-order";
 import { convertedAmount } from "@/utils/numbers";
 
-export const totalStackOrdersDone = (order: StackOrder) =>
-  order.cowOrders.length;
+const noCowOrders = (order: StackOrder) =>
+  !order?.cowOrders || (order.cowOrders && order.cowOrders.length === 0);
+
+export const totalStackOrdersDone = (order: StackOrder) => {
+  if (noCowOrders(order)) return 0;
+
+  return order.cowOrders.length;
+};
 
 export const calculateStackAveragePrice = (order: StackOrder) => {
   let totalExecutedBuyAmount = 0;
   let totalExecutedSellAmount = 0;
 
-  if (order.cowOrders.length === 0) return 0;
+  if (noCowOrders(order)) return 0;
```

Comment:
This looks simpler to me

```suggestion
  if (!order?.cowOrders?.length) return 0;
```

Reply:
I'll test it, it it works, I'm down with this

Reply:
Fixed!